### DISCUSSION TITLE
Does q-virtual-scroll always need "max-height" to be set?

### DISCUSSION THREAD

**Discussion by BenJackGill** [quasarframework/quasar#15899] [General - Components / Directives / etc] [answered]
My experience with `q-virtual-scroll` does not align with the docs. I'm not sure if this should be an issue or a discussion though.

**Problem Summary:** My list will only load a subset when I use a `max-height` value.

**Test 1: With "max-height" - loads a subset**

When I create an `q-virtual-scroll` with a `style="max-height: 300px;"` as per the docs, it will only load a small subset of items on scroll. This is working as expected. See example here: https://codepen.io/thecashbag/pen/rNqRMKE

**Test 2: No "max-height" - loads everything**

But when I remove the `style="max-height: 300px;"` it loads all of the list items at once. See example here: https://codepen.io/thecashbag/pen/rNqRMaQ

**Test 3: With "virtual-scroll-slice-size" and no "max-height" - loads everything**

The docs say this:

> The number of items that will be rendered will be calculated based on the `virtual-scroll-item-size` prop and the size of the scrollable area, but you can fit it to your needs using the `virtual-scroll-slice-size` prop.

But when I have `virtual-scroll-slice-size` set, without `style="max-height: 300px;"`, it still loads the full list. See example here: https://codepen.io/thecashbag/pen/GRYejLy

**Test 4: With "virtual-scroll-slice-size" and "virtual-scroll-item-size" and no "max-height" - loads everything**

The same happens if I have `virtual-scroll-slice-size` and `virtual-scroll-item-size` set, without `style="max-height: 300px;"`, it still loads the full list. See example here:  https://codepen.io/thecashbag/pen/PoyLGvR

**Conclusion and Question:**

The only way I can get it to load a limited subset of the list is to add `style="max-height: 300px;"`.

So does that mean a `max-height` value is always needed to have `q-virtual-scroll` working?

I'm confused by this behaviour because the docs do not make it sound that way.

**Comment by pdanpdan** [comment, marked as answer]
Virtual scroll works by populating items based on container size, so if your element has no size limit it will populate with all items

**Reply by BenJackGill** [reply]
Ok thanks that explains it!

Is there any performance advantage of a `q-virtual-scroll` that loads everything at once vs a normal `q-list`?

Because I have some animations that seem to run smoother when inside a `q-virtual-scroll` even though it is loading everything at once.

**Reply by pdanpdan** [reply]
No, it would be a huge disadvantage :)
It has to do more things to display what amounts to everything
